evalml/data_checks/datetime_format_data_check.py

```diff
@@ -179,71 +195,79 @@ def validate(self, X, y):
             )
             return messages
 
+        # Check if the data is monotonically increasing
+        no_nan_datetime_values = datetime_values.dropna()
+        if not pd.DatetimeIndex(no_nan_datetime_values).is_monotonic_increasing:
+            messages.append(
+                DataCheckError(
+                    message="Datetime values must be sorted in ascending order.",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.DATETIME_IS_NOT_MONOTONIC,
+                ).to_dict()
+            )
+
         col_name = (
             self.datetime_column if self.datetime_column != "index" else "either index"
         )
 
-        nan_columns = datetime_values.isna().any()
-        if nan_columns:
+        ww_payload = infer_frequency(
+            pd.Series(datetime_values),
+            debug=True,
+            window_length=5,
+            threshold=0.8,
+        )
+        inferred_freq = ww_payload[0]
+        debug_object = ww_payload[1]
+        if inferred_freq is not None:
```

Comment:
A little confused why you'd want to return here if there's a detected inferable frequency.  At this point you've determined whether it's monotonic or not and that it has datetime information...

Reply:
If woodwork returns an inferred frequency, it's the same as if the `pd.infer_freq` function returned a frequency. In that case, the datetime column is perfect and we don't need to run the rest of the checks as we know they don't hold.

Now that I talk that through though, I realize we can move that check up higher. We do run `inferred_freq = pd.infer_freq(...)`, so it would make more sense to check if the frequency is perfect up there instead.

Reply:
Thanks for explaining that one to me!